packages/ics721/src/execute.rs

```diff
@@ -385,10 +388,10 @@ where
         let mint = tokens
             .into_iter()
             .map(|Token { id, uri, data }| {
-                // We save token metadata here as, ideally, once cw721
-                // supports on-chain metadata, this is where we will set
-                // that value on the debt-voucher token. Note that this is
-                // set for every token, regardless of if data is None.
+                // Source chain may have provided token metadata, so we save token metadata here
+                // Note, once cw721 doesn't support on-chain metadata yet - but this is where we will set
+                // that value on the debt-voucher token once it is supported.
+                // Also note that this is set for every token, regardless of if data is None.
```

Comment:
This  TOKEN_METADATA part is on receival on target chain, here we store metadata, for later re-transfer. This is not used on mint, since cw721 doesnt support on-chain metadata (yet).